### ISSUE TITLE
systemd-resolved DNS search domain still present after tailscale down

### ISSUE THREAD

**Issue by tavianator** [tailscale/tailscale#1892]
While trying to debug #1889, I noticed this:

```shell
$ tail -n3 /etc/resolv.conf
nameserver 127.0.0.53
options edns0 trust-ad
search localdomain
$ sudo tailscale up
$ tail -n3 /etc/resolv.conf
nameserver 127.0.0.53
options edns0 trust-ad
search localdomain tavianator.gmail.com.beta.tailscale.net
$ sudo tailscale down
$ tail -n3 /etc/resolv.conf
nameserver 127.0.0.53
options edns0 trust-ad
search localdomain tavianator.gmail.com.beta.tailscale.net
```

At this point, DNS queries hang until I restart `systemd-resolved.service`.

This is Arch Linux, tailscale 1.8.1, systemd 248.

**Comment by tavianator** [comment]
Oh I forgot, this computer uses NetworkManager as well, version 1.30.4

**Comment by tavianator** [comment]
Here's the logs from `tailscale down`:

```
May 10 11:16:38 graviton tailscaled[21974]: health("overall"): error: state=Stopped, wantRunning=false
May 10 11:16:38 graviton tailscaled[21974]: control: mapRoutine: paused
May 10 11:16:38 graviton tailscaled[21974]: control: mapRoutine: awaiting unpause
May 10 11:16:38 graviton tailscaled[21974]: magicsock: SetPrivateKey called (zeroed)
May 10 11:16:38 graviton tailscaled[21974]: magicsock: closing connection to derp-1 (zero-private-key), age 4m51s
May 10 11:16:38 graviton tailscaled[21974]: magicsock: 0 active derp conns
May 10 11:16:38 graviton tailscaled[21974]: wgengine: Reconfig: configuring userspace wireguard config (with 0/0 peers)
May 10 11:16:38 graviton tailscaled[21974]: wgengine: Reconfig: configuring router
May 10 11:16:38 graviton tailscaled[21974]: magicsock: ReSTUN("link-change-minor") ignored; stopped, no private key
May 10 11:16:38 graviton tailscaled[21974]: monitor: RTM_DELROUTE: src=, dst=fd7a:115c:a1e0::/48, gw=, outif=8, table=52
May 10 11:16:38 graviton tailscaled[21974]: monitor: RTM_DELROUTE: src=, dst=fd7a:115c:a1e0:ab12:4843:cd96:6267:2c42/128, gw=, outif=8, table=255
May 10 11:16:38 graviton tailscaled[21974]: monitor: RTM_DELROUTE: src=, dst=fd7a:115c:a1e0:ab12:4843:cd96:6267:2c42/128, gw=, outif=8, table=254
May 10 11:16:38 graviton tailscaled[21974]: monitor: RTM_DELROUTE: src=100.103.44.66/0, dst=100.103.44.66/32, gw=, outif=8, table=255
May 10 11:16:38 graviton tailscaled[21974]: wgengine: Reconfig: configuring DNS
May 10 11:16:38 graviton tailscaled[21974]: dns: Set: {DefaultResolvers:[] Routes:map[] SearchDomains:[] Hosts:map[] AuthoritativeSuffixes:[]}
May 10 11:16:38 graviton tailscaled[21974]: dns: Resolvercfg: {Routes:map[] Hosts:map[] LocalDomains:[]}
May 10 11:16:38 graviton tailscaled[21974]: dns: OScfg: {Nameservers:[] SearchDomains:[] MatchDomains:[]}
May 10 11:16:38 graviton tailscaled[21974]: Reconfig(down): interface not ready
May 10 11:16:38 graviton tailscaled[21974]: control: controlclient: restarting map request for "dns" health change to new state: interface not ready
May 10 11:16:38 graviton tailscaled[21974]: health("dns"): error: interface not ready
May 10 11:16:38 graviton tailscaled[21974]: magicsock: ReSTUN("link-change-minor") ignored; stopped, no private key
May 10 11:16:38 graviton tailscaled[21974]: control: HostInfo: {"IPNVersion":"1.8.1-tfbfdcc56e","BackendLogID":"c390b923a4fc93818b9907c126d8e8f9391fd95b0f5be06387f0762db6b43ef1","OS":"linux","OSVersion":"Arch Linux; kernel=5.12.1-arch1-1","Hostname":"graviton","GoArch":"amd64","Services":[{"Proto":"tcp","Port":53,"Description":"systemd-resolved"},{"Proto":"tcp","Port":1716,"Description":"kdeconnectd"},{"Proto":"tcp","Port":5355,"Description":"systemd-resolved"}],"NetInfo":{"MappingVariesByDestIP":false,"HairPinning":false,"WorkingIPv6":false,"WorkingUDP":true,"UPnP":false,"PMP":false,"PCP":false,"PreferredDERP":1,"DERPLatency":{"1-v4":0.01590719,"10-v4":0.073799431,"11-v4":0.136116287,"2-v4":0.064936701,"4-v4":0.09482731,"7-v4":0.178418268,"8-v4":0.081902586,"9-v4":0.040933239}}}
May 10 11:16:39 graviton tailscaled[21974]: dns: error: upstream nameservers not set
May 10 11:16:39 graviton tailscaled[21974]: dns: error: upstream nameservers not set
May 10 11:16:39 graviton tailscaled[21974]: dns: error: upstream nameservers not set
May 10 11:16:40 graviton tailscaled[21974]: dns: error: upstream nameservers not set
May 10 11:16:40 graviton tailscaled[21974]: dns: error: upstream nameservers not set
May 10 11:16:40 graviton tailscaled[21974]: [RATELIMIT] format("dns: error: %v")
May 10 11:16:49 graviton tailscaled[21974]: [RATELIMIT] format("dns: error: %v") (9 dropped)
```

**Comment by tavianator** [comment]
BUG-c390b923a4fc93818b9907c126d8e8f9391fd95b0f5be06387f0762db6b43ef1-20210510190038Z-f45b4605f0d4fba9

**Comment by danderson** [comment]
Thanks. Found the problem, the DNS manager logic looks up the Tailscale interface index by looking for the interface with Tailscale IPs configured on it... Which tailscale down deletes right before calling the DNS manager to update configs. Easy fix, it'll be in 1.8.3. Thanks for the report!
